src/merkle/smt/mod.rs

```diff
@@ -256,7 +257,7 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
     }
 
     /// Apply the prospective mutations computed with [`SparseMerkleTree::compute_mutations()`] to
-    /// this tree.
+    /// this tree. Return reverse mutation set.
```

Comment:
nit: I would make this a single sentence:
```rust
/// Applies the prospective mutations computed with [`SparseMerkleTree::compute_mutations()`] to
/// this tree and returns the reverse mutation set.
```

I would also add a short paragraph (1 - 2 sentences) to explain what "reverse mutation set" is.